src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -68,8 +83,37 @@ export class SponsoredAddonsShelfBase extends React.Component<InternalProps> {
         }),
       );
     }
+
+    this.sendImpressionPing();
+  }
+
+  componentDidUpdate(prevProps: InternalProps) {
+    const { shelfData } = this.props;
+
+    if (prevProps.shelfData !== shelfData) {
+      this.sendImpressionPing();
+    }
   }
 
+  sendImpressionPing = () => {
+    const { _config, _navigator, shelfData } = this.props;
+
+    if (_config.get('enableFeatureUseAdzerkForSponsoredShelf') && shelfData) {
+      const { impressionData, impressionURL } = shelfData;
+
+      if (impressionData && impressionURL) {
+        sendBeacon({
+          _navigator,
+          data: formatDataForPing({
+            data: impressionData,
+            key: 'impression_data',
+          }),
+          urlString: impressionURL,
```

Comment:
I suppose the API is needed in the CSP because this url is absolute. I am wondering if that has any security implications. We could potentially make this URL relative but maybe it's fine as it is currently?

Reply:
We actually already include the apiHost in the `connectSrc` for all other environments. The only change in this patch is that we're adding it to the `development` config, so I don't think there are any security implications.